frappe/model/base_document.py

```diff
@@ -162,7 +162,12 @@ def set(self, key, value, as_value=False):
 		if key in self.ignore_in_setter:
 			return
 
-		if isinstance(value, list) and not as_value:
+		if (
+			not as_value
+			and isinstance(value, list)
+			# if child doctype not found, probably just a property with list as value
+			and self.get_table_field_doctype(key)
```

Comment:
added in `set()` because we can avoid passing the value to `append()`, `extend()` and `_init_child()` unnecessarily if not a child table.

I think it is safe to assume that if `extend()`, `append()` or `_init_child()` is being called directly, the value is supposed to be a child table.